src/modules/Packages.rb

```diff
@@ -2420,6 +2353,62 @@ def remote_x11_packages
 
   private
 
+    # Current package, pattern, product, patch and language selection.
+    #
+    # @return [Hash] selected packages, patterns, products,...
+    def current_proposal
+      selected_packages = Pkg.GetPackages(:selected, false)
+      selected_patterns = Pkg.ResolvableProperties("", :pattern, "").map do |p|
```

Comment:
NP: I am not sure if it is a good idea, but what about having a method like 

```ruby
def fetch_selected(category)
  items = Pkg.ResolvableProperties("", category, "").select { |i| i["status"] == :selected }
  items.map { |i| i["name"] }.sort
end
```

which allows, 

```ruby
def current_proposal
  {
    "packages"  => Pkg.GetPackages(:selected, false).compact.sort,
    "patterns"  => fetch(:pattern),
    "products"  => fetch(:product),
    "patches"   => fetch(:patch),
    "languages" => [Pkg.GetPackageLocale].concat(Pkg.GetAdditionalLocales).compact.sort
  }
end
```

But I said, is just an idea, not sure if good or not.